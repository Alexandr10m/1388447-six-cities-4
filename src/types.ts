export interface LocalOffer {
  isPremium: boolean;
  pictures: string[];
  price: number;
  isFavourite: boolean;
  grade: number;
  title: string;
  type: string;
  bedroom: number;
  maxAdults: number;
  facilities: string[];
  coords: number[];
  locationZoom: number;
  id: number;
  description: string;
  previewImage: string;
  host: Host;
}

export interface CityOffers {
  city: string;
  cityCoords: number[];
  cityZoom: number;
  localOffers: LocalOffer[] | [];
}

export interface Host {
  id: number;
  name: string;
  isPro: boolean;
  avatarUrl: string;
}

export interface author {
  avatarUrl: string;
  email: string;
  id: number;
  isPro: boolean;
  name: string;
}

export interface Review {
  text: string;
  date: InstanceType<typeof Date>;
  id: number;
  grade: number;
  user: author;
}
