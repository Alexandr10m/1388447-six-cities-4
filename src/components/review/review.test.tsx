import * as React from "react";
import * as renderer from "react-test-renderer";
import Review from "./review";
import {Review as ReviewInterface} from "../../types";

const review: ReviewInterface = {
  date: new Date(1),
  grade: 4,
  id: 1,
  text: `We loved it`,
  user: {
    avatarUrl: `https`,
    email: undefined,
    id: 18,
    isPro: true,
    name: `Sophie`,
  },
};


describe(`Snapshot of Review`, () => {
  it(`Review should render correctly`, () => {
    const tree = renderer
      .create(
          <Review
            {...review}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
