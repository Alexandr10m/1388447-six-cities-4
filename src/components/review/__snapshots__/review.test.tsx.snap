// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Review Review should render correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="https"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Sophie
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      We loved it
    </p>
    <time
      className="reviews__time"
      dateTime="1970-01-01"
    >
      January 1970
    </time>
  </div>
</li>
`;
